### PR TITLE
test(NODE-4271): sync fle2 api spec tests

### PR DESCRIPTION
**Pull request by durran**
### Description

Syncs the FLE2 spec tests for DRIVERS-2297

#### What is changing?

Syncs the FLE2 spec tests to this commit: https://github.com/mongodb/specifications/commit/c3784d44555752bb058993e7b0bb51cab5b3b1d5

##### Is there new documentation needed for these changes?

No

#### What is the motivation for this change?

NODE-4271/DRIVERS-2297

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [x] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

**Comment by durran** [comment]
The fle failures in evergreen for event count mismatches are fixed in https://github.com/mongodb/node-mongodb-native/pull/3271

**Review by addaleax — APPROVED** [review]

File: src/encrypter.ts [hunk 1]
```diff
@@ -83,9 +83,6 @@ export class Encrypter {
 
       clonedOptions.minPoolSize = 0;
 
-      clonedOptions.promoteValues = false;
```
Comment (durran) [inline src/encrypter.ts:86]: No longer needed as the state machine in mongodb-client-encryption handles adding of these options.

File: test/tools/spec-runner/index.js [hunk 2]
```diff
@@ -364,6 +363,11 @@ function runTestSuiteTest(configuration, spec, context) {
     [Symbol.for('@@mdb.skipPingOnConnect')]: true
   });
 
+  if (context.requiresCSFLE) {
```
Comment (durran) [inline test/tools/spec-runner/index.js:366]: Always use non-promoted values in the test client when using CSFLE.

File: test/tools/spec-runner/index.js [hunk 3]
```diff
@@ -676,7 +680,7 @@ const kOperations = new Map([
       const encryptedFields = operation.arguments.encryptedFields;
       const session = maybeSession(operation, context);
       return db.dropCollection(collectionName, { session, encryptedFields }).catch(err => {
-        if (err.code !== MONGODB_ERROR_CODES.NamespaceNotFound) {
+        if (!err.message.match(/ns not found/)) {
```
Comment (durran) [inline test/tools/spec-runner/index.js:683]: It's safer to match on strings here as when promoting or not we could be working with a code that is a JS number or `Int32` object. Strings are always strings.

**Comment by durran** [comment]
Also note that fle2-validatorAndPartialFieldExpression.json/yml was part of this PR (https://github.com/mongodb/specifications/pull/1227) but we actually already synced them in: https://github.com/mongodb/node-mongodb-native/commit/4121ec6dd6f2f6550a90796e0c663092527708ef

**Review by nbbeeken — APPROVED** [review]

**Review by dariakp — APPROVED** [review]

**Pull request merged**